easybuild/tools/build_from_specsfile.py

```diff
@@ -0,0 +1,150 @@
+import yaml
+from easybuild.tools.robot import search_easyconfigs
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.base import fancylogger
+
+
+_log = fancylogger.getLogger('specsfile', fname=False)
+
+
+# general specs applicable to all commands
+class Specsfile(object):
+    def __init__(self):
+        self.easybuild_version = None
+        self.robot = False
+        self.software_list = []
+
+    # returns list of all commands - finished
+    def compose_full_paths(self):
+        easyconfigs_full_paths = []
+        for sw in self.software_list:
+            path_to_append = self.get_ec_path(sw)
+
+            if path_to_append is None:
+                continue
+            else:
+                easyconfigs_full_paths.append(path_to_append)
+        return easyconfigs_full_paths
+
+    # single command
+    def get_ec_path(self, sw):
+        full_path = search_easyconfigs(query='%s-%s-%s-%s' % (
+            str(sw.software), str(sw.version), str(sw.toolchain_name), str(sw.toolchain_version)),
+            short=False, filename_only=False,
+            terse=False, consider_extra_paths=True, print_result=False, case_sensitive=False)
+        if len(full_path) == 1:
+            return full_path[0]
+        else:
+            print('%s does not have clearly specified parameters - %s matches found. Skipping. \n' %
+                  (sw.software, str(len(full_path))))
+
+    # todo: flags applicable to all sw (i.e. robot)
+    def get_general_options(self):
+        general_options = {}
+        general_options['robot'] = self.robot
+        general_options['easybuild_version'] = self.easybuild_version
+        return general_options
+
+
+# single sw command
+class SoftwareSpecs(object):
+    def __init__(self, software, version, toolchain, toolchain_version, toolchain_name):
+        self.software = software
+        self.version = version
+        self.toolchain = toolchain
```

Comment:
keeping track of both the toolchain name+version combo and the toolchain name and version separately doesn't make much sense, so drop `self.toolchain` and only use `self.toolchain_version` and `self.toolchain_name`?

Or maybe rework to make `self.toolchain` a dict value with values like `{'name': 'foss', 'version': '2020a'}`, which matches how we usually keep track of toolchains in other places (incl. in easyconfigs).

So:

```python
self.toolchain = {
    'name': toolchain_name,
    'version': toolchain_version,
}